File: Problemset/two-sum/two-sum.js

// @Title: 两数之和 (Two Sum)
// @Author: 846188037@qq.com
// @Date: 2020-05-22 11:53:37
// @Runtime: 140 ms
// @Memory: 34.7 MB

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for(var i=0;i<nums.length-1;i++)
    {
        for(var j=i+1;j<nums.length;j++)
        {
            if(nums[i]+nums[j]==target)
            {
                var ans=[i,j]
                return ans
            }
        }
    }

};

File: Problemset/two-sum/two-sum.go

// @Title: 两数之和 (Two Sum)
// @Author: 846188037@qq.com
// @Date: 2021-04-22 18:03:39
// @Runtime: 4 ms
// @Memory: 3 MB

func twoSum(nums []int, target int) []int {
    Map:=make(map[int]int)
    for k,v:=range nums{
        i,ok:=Map[target-v]
        if ok{
            return []int{k,i}
        }
        Map[v]=k
    }
    return []int{}
}
